gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/MRCompactorAvroKeyJobRunner.java

```diff
@@ -0,0 +1,249 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce.avro;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import gobblin.compaction.mapreduce.MRCompactorJobRunner;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.util.AvroUtils;
+
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaCompatibility;
+import org.apache.avro.Schema.Field;
+import org.apache.avro.SchemaCompatibility.SchemaCompatibilityType;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapred.AvroValue;
+import org.apache.avro.mapreduce.AvroJob;
+import org.apache.avro.mapreduce.AvroKeyOutputFormat;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A subclass of {@link gobblin.compaction.mapreduce.MRCompactorJobRunner} that configures
+ * and runs MR compaction job for Avro data.
+ *
+ * To dedup using entire records set compaction.use.all.attributes=true. Otherwise, a schema needs
+ * to be provided by compaction.avro.key.schema.loc, based on which the dedup is performed.
+ *
+ * @author ziliu
+ */
+public class MRCompactorAvroKeyJobRunner extends MRCompactorJobRunner {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MRCompactorAvroKeyJobRunner.class);
+  private static final String AVRO = "avro";
+
+  public MRCompactorAvroKeyJobRunner(State jobProps, FileSystem fs) {
+    super(jobProps, fs);
+  }
+
+  @Override
+  protected void configureJob(Job job) throws IOException {
+    super.configureJob(job);
+    configureSchema(job);
+  }
+
+  private void configureSchema(Job job) throws IOException {
+    Schema newestSchema = getNewestSchemaFromSource(job);
+    Schema keySchema = getKeySchema(job, newestSchema);
+    AvroJob.setInputKeySchema(job, newestSchema);
+    AvroJob.setMapOutputKeySchema(job, keySchema);
+    AvroJob.setMapOutputValueSchema(job, newestSchema);
+    AvroJob.setOutputKeySchema(job, newestSchema);
+  }
+
+  private Schema getKeySchema(Job job, Schema newestSchema) throws IOException {
```

Comment:
Javadoc. This is the schema used for comparison, but that is not entirely clear.